realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -697,261 +634,133 @@ public void where_afterContainerObjectRemoved() {
-
         assertEquals("Dog@[invalid]", dogs.toString());
     }
 
     @Test
-    public void query() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        RealmList<Dog> dogs = owner.getDogs();
-        Dog firstDog = dogs.where().equalTo("name", "Dog 0").findFirst();
+    public void toString_managedMode() {
+        StringBuilder sb = new StringBuilder("Dog@[");
+        for (int i = 0; i < collection.size() - 1; i++) {
+            sb.append(collection.get(i).row.getIndex());
+            sb.append(",");
+        }
+        sb.append(collection.get(TEST_SIZE - 1).row.getIndex());
+        sb.append("]");
 
-        assertNotNull(firstDog);
+        assertEquals(sb.toString(), collection.toString());
     }
 
     @Test
-    public void allMethods_emptyListInManagedMode() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        testRealm.beginTransaction();
-        owner.getDogs().clear();
-        testRealm.commitTransaction();
+    public void query() {
+        Owner owner = realm.where(Owner.class).findFirst();
+        RealmList<Dog> dogs = owner.getDogs();
+        Dog firstDog = dogs.where().equalTo("name", "Dog 0").findFirst();
 
-        checkMethodsOnEmptyList(testRealm, owner.getDogs());
+        assertNotNull(firstDog);
     }
 
     @Test
     public void clear() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        testRealm.beginTransaction();
-        assertEquals(TEST_OBJECTS, owner.getDogs().size());
+        Owner owner = realm.where(Owner.class).findFirst();
+        realm.beginTransaction();
+        assertEquals(TEST_SIZE, owner.getDogs().size());
         owner.getDogs().clear();
         assertEquals(0, owner.getDogs().size());
-        testRealm.commitTransaction();
+        realm.commitTransaction();
     }
 
     @Test
-    public void clear_NotDeleting() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        testRealm.beginTransaction();
-        assertEquals(TEST_OBJECTS, testRealm.allObjects(Dog.class).size());
+    public void clear_notDeleting() {
+        Owner owner = realm.where(Owner.class).findFirst();
+        realm.beginTransaction();
+        assertEquals(TEST_SIZE, realm.allObjects(Dog.class).size());
         owner.getDogs().clear();
-        assertEquals(TEST_OBJECTS, testRealm.allObjects(Dog.class).size());
-        testRealm.commitTransaction();
-    }
-
-    @Test
-    public void contains() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        Dog dog = owner.getDogs().get(0);
-        assertTrue("Should contain a particular dog.", owner.getDogs().contains(dog));
+        assertEquals(TEST_SIZE, realm.allObjects(Dog.class).size());
+        realm.commitTransaction();
     }
 
-    /**
-     * Test to see if a particular item that does exist in the same Realm does not
-     * exist in a query that excludes said item.
-     */
     @Test
-    public void contains_sameRealmNotContained() {
```

Comment:
Yes, this test didn't make much sense, since it more about how RealmQuery works than the Collection